brem/AsyncCopyOverSSH.py

```diff
@@ -0,0 +1,186 @@
+from PySide2 import QtCore
+from eqt.threading import Worker
+import paramiko
+from brem import BasicRemoteExecutionManager
+import os, ntpath, posixpath
+import socket
+
+class RemoteAsyncCopyOverSSHSignals(QtCore.QObject):
+    status = QtCore.Signal(tuple)
+    job_id = QtCore.Signal(int)
+class AsyncCopyOverSSH(object):
+    '''Class to handle async copy of files over SSH'''
+    def __init__(self, parent=None):
+
+        self.internalsignals = RemoteAsyncCopyOverSSHSignals()
+        self.threadpool = QtCore.QThreadPool()
+        self.logfile = 'AsyncCopyFromSSH.log'
+        self._worker = None
+        self.dest_fname = None
+        
+    def SetRemoteDir(self, dirname):
+        self.remotedir = dirname
+
+    def SetFileName(self, filename):
+        self.filename = filename
+    
+    def SetDestinationFileName(self, fname):
+        self.dest_fname = fname
+
+    def SetLocalDir(self, dirname):
+        self.localdir = dirname
+
+    def SetCopyToRemote(self):
+        self.direction = 'to'
+
+    def SetCopyFromRemote(self):
+        self.direction = 'from'
+    
+    def SetRemoteOS(self, value):
+        allowed_os = ['Windows', 'POSIX']
+        if value in allowed_os:
+            self.remote_os = value
```

Comment:
the reason to set `remote_os` is to use `posixpath` or `ntpath`. 
Probably this class could have a default `POSIX` and this method should just configure the `remotepath` that can be used later:

```python

if value == 'POSIX':
    self.remotepath = posixpath
elif value == 'Windows':
    self.remotepath = ntpath
else:
    raise ValueError('Expected value in {}. Got {}'.format(['POSIX','Windows'], value))

```